tests/test_well_object_funcs.py

```diff
@@ -0,0 +1,208 @@
+import os
+import pandas as pd
+import numpy as np
+import pytest
+
+from resqpy.well.well_object_funcs import add_wells_from_ascii_file, well_name, add_blocked_wells_from_wellspec, add_logs_from_cellio, lookup_from_cellio
+import resqpy.well
+from resqpy.grid import RegularGrid
+
+
+def test_add_wells_from_ascii_file(example_model_and_crs, caplog):
+
+    # --------- Arrange ----------
+    model, crs = example_model_and_crs
+    crs_uuid = crs.uuid
+    ascii_file = os.path.join(model.epc_directory, 'ascii.dat')
+    well_name = 'Well_1'
+
+    source_df = pd.DataFrame([[well_name, 200, 1, 1, 150], [well_name, 250, 2, 2, 200], [well_name, 300, 2, 3, 250]],
+                             columns = ['WELL', 'MD', 'X', 'Y', 'Z'])
+    with open(ascii_file, 'w') as fp:
+        for col in source_df.columns:
+            fp.write(f' {col:>6s}')
+        fp.write('\n')
+        for row_index in range(len(source_df)):
+            row = source_df.iloc[row_index]
+            for col in source_df.columns:
+                if col == 'WELL':
+                    fp.write(f' {(row[col]):6s}')
+                else:
+                    fp.write(f' {row[col]:6.2f}')
+            fp.write('\n')
+
+    well_col_name = 'Wellname'
+    with pytest.raises(AssertionError) as exc:
```

Comment:
Does the testing include a successful execution of the add_wells_from_ascii_file() function, as well as this failure mode test.

Reply:
now it does :D